File: cs/Minsk/CodeAnalysis/Syntax/AssignmentExpressionSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public sealed class AssignmentExpressionSyntax : ExpressionSyntax
{
    public AssignmentExpressionSyntax(SyntaxToken identifierToken, SyntaxToken equalsToken, ExpressionSyntax expression)
    {
        IdentifierToken = identifierToken;
        EqualsToken = equalsToken;
        Expression = expression;
    }

    public SyntaxToken IdentifierToken { get; }
    public SyntaxToken EqualsToken { get; }
    public ExpressionSyntax Expression { get; }
    public override SyntaxKind Kind => SyntaxKind.AssignmentExpression;

    public override IEnumerable<SyntaxNode> Children =>
        new SyntaxNode[] { IdentifierToken, EqualsToken, Expression };
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/AssignmentExpressionSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class AssignmentExpressionSyntax(
    val identifierToken: SyntaxToken,
    val equalsToken: SyntaxToken,
    val expression: ExpressionSyntax,
) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.AssignmentExpression
    override val children: List<SyntaxNode>
        get() = listOf(identifierToken, equalsToken, expression)
}
